master/master.cfg

```diff
@@ -100,31 +100,34 @@ HALIDE_BRANCHES = {HALIDE_MAIN: VersionedBranch(ref='master', version=Version(12
-    # ('linux-worker-2', WorkerConfig(max_builds=2, arch='x86', bits=[32, 64], os='linux')),
-    # ('linux-worker-3', WorkerConfig(max_builds=2, arch='x86', bits=[32, 64], os='linux')),
-    ('linux-worker-4', WorkerConfig(max_builds=4, arch='x86', bits=[32, 64], os='linux')),
-    ('mac-worker-1', WorkerConfig(max_builds=2, arch='x86', bits=[64], os='osx')),
-    ('mac-arm-worker-1', WorkerConfig(max_builds=2, arch='arm', bits=[64], os='osx')),
-    ('arm32-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('arm32-linux-worker-2', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('arm64-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[64], os='linux')),
-    ('arm64-linux-worker-2', WorkerConfig(max_builds=1, arch='arm', bits=[64], os='linux')),
-    # TODO: this bot is configured to handle both arm64 and arm32 builds. For now, just running arm32.
-    # ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32, 64], os='linux')),
-    ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('win-worker-1', WorkerConfig(max_builds=1, arch='x86', bits=[32, 64], os='windows')),
-    ('win-worker-2', WorkerConfig(max_builds=1, arch='x86', bits=[32, 64], os='windows')),
+    ('linux-worker-1', WorkerConfig(max_builds=4, j=8, arch='x86', bits=[32, 64], os='linux')),
+    ('linux-worker-4', WorkerConfig(max_builds=4, j=8, arch='x86', bits=[32, 64], os='linux')),
+    ('mac-worker-1', WorkerConfig(max_builds=2, j=8, arch='x86', bits=[64], os='osx')),
+    ('mac-arm-worker-1', WorkerConfig(max_builds=2, j=8, arch='arm', bits=[64], os='osx')),
+    ('arm32-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('arm32-linux-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('arm64-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[64], os='linux')),
+    ('arm64-linux-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[64], os='linux')),
+    ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('win-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='x86', bits=[32, 64], os='windows')),
+    ('win-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='x86', bits=[32, 64], os='windows')),
 ]
 
 # The 'workers' list defines the set of recognized buildworkers. Each element is
 # a Worker object, specifying a unique worker name and password.  The same
 # worker name and password must be configured on the worker.
-c['workers'] = [Worker(n, WORKER_SECRET, max_builds=cfg.max_builds) for n, cfg in _WORKERS]
+c['workers'] = [Worker(n,
+                       WORKER_SECRET,
+                       max_builds=cfg.max_builds,
+                       properties={'WORKER_BUILD_PARALLELISM': cfg.j}) for n, cfg in _WORKERS]
```

Comment:
I like that this gives us a chance to set the `WORKER_BUILD_PARALLELISM` property through other means, eg. via the `SetPropertyFromCommand` step.